src/match/createMatch.ts

```diff
@@ -14,6 +14,15 @@ export async function createMatch(
   matchType: MatchType,
   matchStage: MatchStage
 ) {
+  const isVolunteerAvailable = checkVolunteerAvailability(
+    volunteerAvailability.current_matches,
+    volunteerAvailability.max_matches
+  );
+
+  if (!isVolunteerAvailable) {
+    throw new Error("Volunteer is not available for a new match.");
+  }
+
```

Comment:
Acho que não podemos deixar esse check aqui senão os matches manuais não vão funcionar para casos específicos de voluntárias que, em teoria, não estão disponíveis mas as meninas gostariam de fazer um match manual mesmo assim (porque elas conhecem a voluntária e tal).

Mas acho que ele também não resolveria o problema ):
Porque a questão é que, atualmente, a lambda está processando dois pedidos de acolhimento ao mesmo tempo, em vez de um de cada vez

Reply:
Hmmm, tem razão em relação ao match manual! tô vendo uma forma de resolver isso sem precisar separar uma função createMatch só pro match manual.

O lance de fazer essa verificação é que a gente garantiria que ela tá sendo feita dentro da função createMatch, mas **fora** da constante match que de fato cria o match. Pq assim o que parece é que a voluntária tá sendo atualizada pra indisponível mas ainda assim o match é criado. Vê essa parte do código e me diz se faz sentido pfvr?

se for isso, o ideal seria separar a lógica do match padrão e match manual, posso fazer isso

Reply:
obs: não consegui ver outra forma de resolver ainda